### PR TITLE
Implement sourceRelativePath in manifest.json

### PR DESCRIPTION
**Pull request by qinezh**
@chenkennt @vwxyzh @superyyrrzz @hellosnow @ansyral @herohua

### CODE REVIEW

**Comment by vwxyzh** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Plugins/Manifest.cs [hunk 1]
```diff
@@ -18,6 +18,10 @@ public class Manifest
         [JsonProperty("homepages")]
         public List<HomepageInfo> Homepages { get; set; }
 
+        [YamlMember(Alias = "sourceBasePath")]
+        [JsonProperty("source_base_path")]
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Plugins/Manifest.cs]: >source_base_path [](start = 23, length = 16)

have a question, why not use "sourceBasePath" to be consitent with yaml name and below xrefmap?

File: src/Microsoft.DocAsCode.Build.ResourceFiles/ResourceDocumentProcessor.cs [hunk 2]
```diff
@@ -75,11 +75,19 @@ public override FileModel Load(FileAndType file, ImmutableDictionary<string, obj
...
             var filePath = Path.Combine(file.BaseDir, file.File);
             var repoDetail = GitUtility.GetGitDetail(filePath);
-            var displayLocalPath = repoDetail?.RelativePath ?? Path.Combine(file.BaseDir, file.File).ToDisplayPath();
+            string displayLocalPath = null;
+
+            object baseDirectory;
+            if (metadata.TryGetValue("baseRepositoryDirectory", out baseDirectory))
+            {
+                displayLocalPath = PathUtility.MakeRelativePath((string)baseDirectory, file.FullPath);
+            }
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.ResourceFiles/ResourceDocumentProcessor.cs]: to reduce duplicate code, how about move to SingleDocumentBuilder line 392, after load then set LocalPathFromRoot property?

File: src/Microsoft.DocAsCode.Plugins/Manifest.cs (hunk 1 above)
Comment (qinezh) [inline src/Microsoft.DocAsCode.Plugins/Manifest.cs]: to be consistent with `relative_path` in manifest.json

File: src/Microsoft.DocAsCode.Build.ResourceFiles/ResourceDocumentProcessor.cs (hunk 2 above)
Comment (qinezh) [inline src/Microsoft.DocAsCode.Build.ResourceFiles/ResourceDocumentProcessor.cs]: Seems that `FileModel.Uids` need the `LocalPathFromRoot`

**Comment by superyyrrzz** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.ResourceFiles/ResourceDocumentProcessor.cs [hunk 3]
```diff
@@ -75,11 +75,19 @@ public override FileModel Load(FileAndType file, ImmutableDictionary<string, obj
 
             var filePath = Path.Combine(file.BaseDir, file.File);
             var repoDetail = GitUtility.GetGitDetail(filePath);
-            var displayLocalPath = repoDetail?.RelativePath ?? Path.Combine(file.BaseDir, file.File).ToDisplayPath();
+            string displayLocalPath = null;
+
+            object baseDirectory;
+            if (metadata.TryGetValue("baseDirectory", out baseDirectory))
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.ResourceFiles/ResourceDocumentProcessor.cs]: >baseDirectory [](start = 38, length = 13)

_baseDirectory?

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**